lib/puppet/functions/extlib/cidr_to_range.rb

```diff
@@ -0,0 +1,21 @@
+# Imported by Tim 'bastelfreak' Meusel into voxpupuli/extlib because Yelp/netstdlib got abandoned
+#
+# @summary Converts an CIDR address of the form 192.168.0.1/24 into a range of IP address excluding the network and broadcast addresses.
+#
+Puppet::Functions.create_function(:'extlib::cidr_to_range') do
+  # @param ip IPv6 or IPv4 address in CIDR notation
+  # @return IPv6 or IPv4 ip range without net/broadcast
+  # @example calling the function
+  #   extlib::cidr_to_range('127.0.0.1/8')
+  dispatch :cidr_to_range do
+    param 'Variant[Stdlib::IP::Address::V4::CIDR,Stdlib::IP::Address::V6::CIDR]', :ip
+    return_type 'Variant[Array[Stdlib::IP::Address::V4::Nosubnet],Array[Stdlib::IP::Address::V6::Nosubnet]]'
+  end
+
+  def cidr_to_range(ip)
+    ips = IPAddr.new(ip).to_range.map(&:to_s)
```

Comment:
Also, this can be slow and memory consuming for huge IPv6 networks like a /64 (or larger). Why would you want this?

Reply:
yes, depending on the subnet size it might take some time.

---

I know at least one person that uses https://github.com/Yelp/puppet-netstdlib and I require those functions as well. Yelp abandoned the module and I had the impression it's a good idea to add them to extlib. I thought for a hand full of short functions it doesnt make much sense to maintain a whole module.

Reply:
what is this used for? eespecially in the case of IPv6?

Reply:
Yeah, not sure why you would want to enumerate every address of a subnet (you're definitely not doing this for most IPv6 networks), but I think it would be more useful to have a function that gets one of the addresses in a subnet as I mentioned here: https://github.com/voxpupuli/puppet-extlib/pull/165#issuecomment-703329300

You could have a loop around such a function if you really wanted to do something with every address.

Reply:
this would make sense if we returned lazy structures